doc/userguide/configuration/suricata-yaml.rst

```diff
@@ -1071,6 +1071,14 @@ percent of the 10000 flows is completed).
   emergency_recovery: 30                  #Percentage of 1000 prealloc'd flows.
   prune_flows: 5                          #Amount of flows being terminated during the emergency mode.
 
+If aggressive flow pruning in emergency-mode is not desired, it can be disabled by
+configuring flow.force_reuse.
```

Comment:
I have a few questions. Could you please tell:
1. what issues were seen that led to the need of this feature?
2. the impact of this setting being `true` in an actual emergency mode. AFAI understand the engine will already be under pressure so we put more pressure on it of creating new flows?

Thanks a lot!

Reply:
Good questions, I have also updated the original feature request on redmine to try to answer these.

1.
The issue was HTTP packets, from a previously blocked TCP connection, getting past IPS during high traffic load. This occurred when a blocked TCP connection's flow was re-used and subsequent packets from the previously blocked TCP connection were not categorized as HTTP.

This was only observed in emergency mode, as the flow timeouts are much shorter.

The solution proposed is to not re-use potentially active flows in emergency mode.

2.
This setting means that new flows will be dropped in emergency mode, until older flows have timed out.

Reply:
Thank you very much for explaining. Is it possible for you to share pcap and rules of your scenario?
I'd expect an active flow to not be reused indeed .

Reply:
In emergency mode (when the memcap has been exceeded), FlowGetUsedFlow() disregards timeouts. Doesn't this lead to active flows being reused in some cases?

The traffic was generated by Spirent, I will attempt to get a pcap that can reproduce the issue.

Reply:
> In emergency mode (when the memcap has been exceeded), FlowGetUsedFlow() disregards timeouts. Doesn't this lead to active flows being reused in some cases?
> 

hmm since it does lead to active flows being reused as your usecase shows, I'd say you're right but I expected any active flows to be rejected w an increment in the `flow.get_used_eval_reject` counter in the stats. Maybe there's a reason active flows are allowed to be reused that I don't see yet. 🤔 

> The traffic was generated by Spirent, I will attempt to get a pcap that can reproduce the issue.

Thank you very much! I'm doing some research on internal Flow handling. That'll be very helpful! 🙇🏽‍♀️ 

Note: I'm not holding this PR from merge w the conversation here, so, feel free to open a new one w the requested changes by Victor and you can share the pcap when you can.